datalad/core/distributed/push.py

```diff
@@ -266,6 +265,23 @@ def __call__(
             for i, ds in pbars.items():
                 log_progress(lgr.info, i, 'Finished push of %s', ds)
         if not matched_anything:
+            potential_remote = False
+            if not to and path:
```

Comment:
```suggestion
            if not to and len(paths) == 1:
```
This is a readily available list of positional args, and we would only have an idea, if there is exactly one arg that may turn out to match a sibling name. We can rule out all other cases here and save cycles.